sidecar/internal/reconciler/bucket_test.go

```diff
@@ -112,6 +113,7 @@ func TestBucketReconciler_Reconcile(t *testing.T) {
 		defer cleanup()
 		require.NoError(t, err)
 		go serve()
+		time.Sleep(time.Millisecond) // Let goroutine start
```

Comment:
I have seen these tests flake a few times, and I'm glad to have a suggested fix.
However, `time.Sleep()` is not a robust way to resolve race conditions in my experience. When the CI test job is scheduled on a full runner, things tend to take longer as the various threads on the system vie for CPU time. In such cases, we can expect tail latencies longer than a millisecond I would guess. 

I would prefer to deal with occasional test flakes in the short term and try to find a metric that we can check after calling `go serve()` to make sure the server is ready for the test to proceed, regardless of whether it takes a few nanoseconds or tens of milliseconds.

Reply:
Yeah, I don't think this is a good idea. We should probably have some retry or signal coming from the goroutine, that the server is ready to serve the incoming calls.